cpp/include/raft/sparse/linalg/norm.cuh

```diff
@@ -66,6 +67,37 @@ void csr_row_normalize_max(const int* ia,  // csr row ind array (sorted by row)
   detail::csr_row_normalize_max(ia, vals, nnz, m, result, stream);
 }
 
+/**
+ * @brief Compute row-wise norm of the input matrix and perform fin_op lambda
+ *
+ * Row-wise norm is useful while computing pairwise distance matrix, for
+ * example.
+ * This is used in many clustering algos like knn, kmeans, dbscan, etc...
+ *
+ * @tparam Type the data type
+ * @tparam Lambda device final lambda
+ * @tparam IdxType Integer type used to for addressing
+ * @param dots the output vector of row-wise dot products
+ * @param ia the input matrix row pointers
+ * @param data the input matrix nnz data
+ * @param N number of rows of data
+ * @param type the type of norm to be applied
+ * @param stream cuda stream where to launch work
+ * @param fin_op the final lambda op
+ */
+template <typename Type, typename IdxType = int, typename Lambda = raft::identity_op>
+void rowNormCsr(Type* dots,
```

Comment:
- How does the performance relate to `csr_row_normalize_l1` (and `.._max`)?
- Any other differences (apart from implementation detail)?